### DISCUSSION TITLE
Solid meshing algorithms

### DISCUSSION THREAD

**Discussion by CharlesJQuarra** [libfive/libfive#466] [Q&A]
Hello, first of all, thanks for this amazing beautiful library

While browsing the documentation and examples, I couldn't find any mention of solid meshing. Is my understanding that the meshing algorithms currently implemented are aimed at rendering and not 3D modeling per se. Am I missing something or is this in the roadmap?

**Comment by mkeeter** [comment]
I'm not sure what you mean by "solid meshing".  Are you talking about generating a tet mesh?

The rendering algorithms generate a triangle (surface) mesh, which is _mostly_ suitable for 3D modeling applications.

To be specific, the output meshes are:
- watertight
- manifold
- preserving of sharp features

They are **not**:
- preserving of thin features
- guaranteed free of self-intersections

**Comment by doug-moen** [comment]
I understand this question to be about Finite Element Modelling for analyzing and simulating engineering structures (using tet meshes). I'm just a bystander, but my impression is that solid meshing is not on the roadmap.

Libfive is a "solid modelling" tool because it represents 3D solid geometric objects using signed distance fields, which is a volumetric representation. It converts the SDF to a triangle mesh (boundary representation) for the purpose of rendering on a display, and for exporting mesh files, which can then be used by other programs, which includes 3D printing. Matt and I are both involved in the 3D printer community.
